### PR TITLE
Windows use 0 dataloader workers.

### PR DESCRIPTION
**Pull request by coledie**
- [ x] Forked repo is synced with upstream -> github shows no code delta outside of the desired.
    https://docs.github.com/en/pull-requests/collaborating-with-pull-requests/working-with-forks/syncing-a-fork
- [x] Tests are passing? https://github.com/microsoft/DeepGNN/blob/main/CONTRIBUTING.md#run-tests
- [x] Changelog and documentation updated.
- [x] PR is labeled using the label menu on the right side.

Previous Behavior
----------------
#83 

Error with run.sh scripts using FileNodeSampler on windows caused by non-zero dataloader workers.

New Behavior
----------------
When using windows, 0 dataloader workers will be used by default to avoid fork.

### CODE REVIEW

File: src/python/deepgnn/pytorch/training/factory.py [hunk 1]
```diff
@@ -117,7 +118,7 @@ def run_dist(
 
     num_workers = (
         0
-        if issubclass(dataset.sampler_class, (GENodeSampler, GEEdgeSampler))
+        if issubclass(dataset.sampler_class, (GENodeSampler, GEEdgeSampler)) or platform.system() == "Windows"
```
Comment (mortid0) [inline src/python/deepgnn/pytorch/training/factory.py:121]: Do we need to run examples in CI on windows to trigger this behavior?

**Review by mortid0 — APPROVED** [review]

File: src/python/deepgnn/pytorch/training/factory.py (hunk 1 above)
Comment (coledie) [reply src/python/deepgnn/pytorch/training/factory.py:121]: Yes the original bug has been here for all deepgnn package versions. Should we add powershell run files to all the examples or install something so we can use bash in windows?
